### PR TITLE
chore(web): list field

### PR DESCRIPTION
**Pull request by jashanbhullar**
# Overview
List field

## What I've done
- created list field with basic logic
- API integrated 
- Updated PropertyField/index.tsx to accommodate for list fields 
- Updated number field to handle NaN values

## What I haven't done
- You tell me 

## How I tested
- stories 
- Editor Ui

## Which point I want you to review particularly
- PropertyField Integration 

## Memo
- NA

### CODE REVIEW

File: web/src/beta/components/fields/ListField/groupsObject.js
Comment (KaWaite) [inline web/src/beta/components/fields/ListField/groupsObject.js:1]: Was this just for testing purposes? Please delete

File: web/src/beta/components/fields/ListField/sceneObject.js
Comment (KaWaite) [inline web/src/beta/components/fields/ListField/sceneObject.js:1]: Was this just for testing purposes? Please delete

File: web/src/beta/lib/lexical/RichTextEditor/index.tsx
Comment (KaWaite) [inline web/src/beta/lib/lexical/RichTextEditor/index.tsx:1]: This is unrelated to this PR, right?

File: web/src/beta/components/fields/PropertyFields/index.tsx
Comment (KaWaite) [inline web/src/beta/components/fields/PropertyFields/index.tsx:1]: Sorry anything that you added in this file that is used ONLY in the list field, please move it in to that component. It seems most/all just need certain IDs that you can pass instead.
Of course I understand there is some refactoring/etc that you and I discussed that we will do later, but please do the above so at least we keep things better organized and the refactoring/clean up we do later will be easier 🙏🏻

File: web/src/beta/components/fields/ListField/sceneObject.js
Comment (jashanbhullar) [reply web/src/beta/components/fields/ListField/sceneObject.js:1]: Yes. Deleted.

File: web/src/beta/lib/lexical/RichTextEditor/index.tsx
Comment (jashanbhullar) [reply web/src/beta/lib/lexical/RichTextEditor/index.tsx:1]: Yea, i was testing something. Accidental commit, removed now!

File: web/src/beta/components/fields/PropertyFields/index.tsx
Comment (jashanbhullar) [reply web/src/beta/components/fields/PropertyFields/index.tsx:1]: Done refactor as discussed

File: web/src/beta/components/fields/PropertyFields/index.tsx [hunk 1]
```diff
@@ -15,119 +18,174 @@ import useHooks from "./hooks";
...
+                id: i.id,
+                value: title as string,
+              };
+            })
+        : [],
+    [isList, item],
+  );
+
+  const showFields = useMemo(() => {
+    return isList ? (selected ? item.items.find(({ id }) => id == selected) : false) : true;
```
Comment (KaWaite) [inline web/src/beta/components/fields/PropertyFields/index.tsx:72]: 90% of the time triple equals is desired 👍🏻 
So should be `id === selected`

**Review by KaWaite — APPROVED** [review]

File: web/src/beta/components/fields/PropertyFields/index.tsx (hunk 1 above)
Comment (jashanbhullar) [reply web/src/beta/components/fields/PropertyFields/index.tsx]: Done Done.
